nemo_aligner/utils/trt_llm.py

```diff
@@ -1,136 +1,111 @@
-                use_refit=True,
-            )
+                nemo_model = model, 
+                nemo_model_config = self.cfg, 
+                trt_model_type = self.trt_model_type,
+                tokenizer = self.tokenizer,
+                max_input_len=self.max_context_length,
+                max_output_len=self.max_generation_length,
+                max_batch_size=self.generation_batch_size,
+                reshard_model=self.reshard_model)
             self._trtllm_model_compiled = True
         else:
             self.trt_llm_exporter.refit(
-                nemo_model=model, nemo_model_config=self.cfg,
+                nemo_model = model, 
+                nemo_model_config = self.cfg, 
             )
 
-    def generate(self, inputs):
-        stop_words = self.stop_words
-        output_ids = self.forward(inputs, stop_words)
-
-        if output_ids is not None:
-            mbs = output_ids.shape[0]
-            if mbs == 1:
-                output_ids = output_ids.view([1, output_ids.shape[-1]])
-            else:
-                output_ids = output_ids.squeeze()
-            output_ids = output_ids.to(torch.int64)
-
-        output_ids = broadcast_2d_tensor_within_mp(output_ids, dtype=torch.long)
-        print("### RANK", torch.distributed.get_rank(), output_ids)
-
-        sentences = [self.tokenizer.ids_to_text(output.tolist()) for output in output_ids]
-        output_ids = output_ids.tolist()
-
-        output = {
-            "token_ids": output_ids,
-            "sentences": sentences,
-        }
 
-        return output
+    def generate(self, inputs):
+        prompt_tokens, prompt_lengths = inputs
 
-    def forward(self, inputs, stop_ids):
-        from nemo.export.trt_llm.tensorrt_llm_run import tensorrt_llm_worker_context
+        batch_input_ids = []
+        for idx in range(prompt_tokens.shape[0]):
+            batch_input_ids.append(prompt_tokens[idx][0:prompt_lengths[idx]].cpu())
 
-        decoder = tensorrt_llm_worker_context.decoder
-        from tensorrt_llm.runtime import ModelRunner
+        output_dict = self.trt_llm_exporter.model_runner.generate(
+            batch_input_ids=batch_input_ids,
+            sampling_config=self.sampling_config,
+            streaming=False)
 
-        self.model_runner = ModelRunner(
-            session=decoder,
-            max_batch_size=self.generation_batch_size,
-            max_input_len=self.max_context_length,
-            max_seq_len=4096,  # TODO: fix
-            max_beam_width=1,
-        )
+        output_ids = output_dict['output_ids']
+        resp_lens = output_dict['sequence_lengths'].squeeze()
 
-        prompt_tokens, prompt_lengths = inputs
+        # remove beam dim from output_ids: [mbs, beam_dim, sequence len]
+        mbs = output_ids.shape[0]
+        if mbs == 1:
+            output_ids = output_ids.view([1,output_ids.shape[-1]])
+        else:
+            output_ids = output_ids.squeeze()
```

Comment:
can't you do `torch.atleast_2d(output_ids).long()` here?

Reply:
From pytorch doc it says ` Input tensors with two or more dimensions are returned as-is.` and output_ids is 3d